syntax = "proto3";

import "google/protobuf/empty.proto";

package ou_pokemon;

service PokemonOUGame
{
    rpc Board(Player) returns (InitialMoves){}
    rpc Checkboard(google.protobuf.Empty) returns (MoveDecision) {}
    rpc MoveTrainer(TMove) returns (Message){}
    rpc Trainer(TrainerName) returns (TrainerInfo){}
    rpc TrainerPath(google.protobuf.Empty) returns (MoveList) {}
    rpc Captured(google.protobuf.Empty) returns (Message) {}
    rpc MovePokemon(PMove) returns (Message){}
    rpc Pokedex(google.protobuf.Empty) returns (PokemonList) {}
    rpc PokemonPath(google.protobuf.Empty) returns (MoveList) {}
}

message Empty {

}
message InitialMoves{
    string exist=1;
}

message PokemonName {
string pname = 1;
}

message PMove {
    string hostname = 1;
}

message TMove {
    string hostname = 1;
}

message Player {
    string player = 1;
}

message Feedback {
    int32 rowNum = 1;
    int32 columnNum = 2;
}

message BoardConfig {
    repeated int32 pokemonRow = 1;
    repeated int32 pokemonColumn = 2;
    repeated int32 trainerRow = 3;
    repeated int32 trainerColumn = 4;
}

message MoveDecision {
    string move = 1;
}

message TrainerName {
    string tname = 1;
}

message TrainerInfo {
    string tname = 1;
    string pname =2;
    int32 row = 3;
    int32 column = 4;
}

message MoveList {
    repeated MoveDecision movelist = 1;
}

message Message {
    int32 count = 1;
}

message PokemonList {
    repeated PokemonName name = 1;
}
